internal/baking/template_variables_service.go

```diff
@@ -45,3 +39,19 @@ func (s TemplateVariablesService) FromPathsAndPairs(paths []string, pairs []stri
 
 	return variables, nil
 }
+
+func parseVariablesFromFile(fs billy.Basic, path string, variables map[string]interface{}) error {
```

Comment:
Could this return `variables` instead? It would be nice for the side effects to be contained.

Reply:
This allows variables to be set by some previous config then overwritten by the file contents. I think we use this functionality somewhere in this code but am not sure.